SwiftCompilerSources/CMakeLists.txt

```diff
@@ -154,6 +153,17 @@ function(add_swift_compiler_modules_library name)
   endforeach()
 
   # Create a static library containing all module object files.
+  #
+  # Xcode does not compile libraries that contain only object files.
+  # Therefore, it fails to create the static library. As a workaround,
+  # we create just a dummy source file.
+  if (XCODE)
+    set(dummy_source_file "${build_dir}/dummy.cpp")
```

Comment:
I think I would be happier with a random SHA sum as the filename rather than something which could collide.  What happens with only objects in the archive?

Reply:
I found a more elegant solution, that works too: creating an empty script phase. That is probably nicer than creating a dummy file. (And if we generated a random SHA sum for the dummy file name every time that CMake runs, wouldn't that cause dummy files to pile up in the build dir?)

Without a dummy file or script phase, Xcode would completely ignore the library target, not even mentioning it in the logs. (It does appear in the Xcode GUI, but is never built.)

Reply:
I wasn't suggesting a random SHA sum :) You could generate a SHA sum of the target name.  The empty files in the build tree don't really matter - they would get cleaned up when you clean up the build tree.